Okay, no wonder calling `module('unload', 'cray-libsci')` is slow, it actually spins up a full Python process...

```
module unload cray-libsci 2>&1 >/dev/null;
export SPACK_NEW_LD_LIBRARY_PATH="$LD_LIBRARY_PATH";LD_LIBRARY_PATH="$SPACK_LD_LIBRARY_PATH"
/path/to/bin/python3 -c "import os;import json;print(json.dumps(dict(os.environ)))";
echo "${SPACK_NEW_LD_LIBRARY_PATH:-SPACKIGNORE}"
```

:(.

Let me add that conditional for module unload cray-libsci then.

(@becker33 note that the above generated shell script does not pass shellcheck validation: https://github.com/koalaman/shellcheck/wiki/SC2069)

--- a/lib/spack/spack/build_environment.py
+++ b/lib/spack/spack/build_environment.py
@@ -147,6 +147,13 @@ class MakeExecutable(Executable):
         return super(MakeExecutable, self).__call__(*args, **kwargs)
 
 
+def _on_cray():
+    hostarch = arch.Arch(arch.platform(), 'default_os', 'default_target')
+    on_cray = str(hostarch.platform) == 'cray'
+    using_cnl = re.match(r'cnl\d+', str(hostarch.os))
+    return on_cray, using_cnl
+
+
 def clean_environment():
     # Stuff in here sanitizes the build environment to eliminate
     # anything the user has set that may interfere. We apply it immediately
@@ -181,9 +188,7 @@ def clean_environment():
     # interference with Spack dependencies.
     # CNL requires these variables to be set (or at least some of them,
     # depending on the CNL version).
-    hostarch = arch.Arch(arch.platform(), 'default_os', 'default_target')
-    on_cray = str(hostarch.platform) == 'cray'
-    using_cnl = re.match(r'cnl\d+', str(hostarch.os))
+    on_cray, using_cnl = _on_cray()
     if on_cray and not using_cnl:
         env.unset('CRAY_LD_LIBRARY_PATH')
         for varname in os.environ.keys():
@@ -804,7 +809,9 @@ def setup_package(pkg, dirty, context='build'):
     # kludge to handle cray libsci being automatically loaded by PrgEnv
     # modules on cray platform. Module unload does no damage when
     # unnecessary
-    module('unload', 'cray-libsci')
+    on_cray, _ = _on_cray()
+    if on_cray:
+        module('unload', 'cray-libsci')
 
     if pkg.architecture.target.module_name:
         load_module(pkg.architecture.target.module_name)

